Make API docker image public in docker-compose.yml

MVrachev opened the pull request:
Make API docker image public in docker-compose.yml.

Close #55 

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

kairoaraujo commented:
Do you think it is possible to have an automatic check if the correct entry is in the file when we have PR?

rdimitrov commented:
> Do you think it is possible to have an automatic check if the correct entry is in the file when we have PR?

Maybe having something like this in our CI?
 
`egrep -w 'image: repository-service-tuf-api|image: repository-service-tuf-worker|image: repository-service-tuf-cli' docker-compose.yml`

rdimitrov commented on .github/workflows/ci.yml:
```diff
@@ -15,6 +15,11 @@ jobs:
         python-versions: [ "3.9", "3.10", ]
 
     steps:
+    - name: Check if any local image is used in docker-compose.yml
+      run: |
+        egrep -w 'image: repository-service-tuf-api|image: repository-service-tuf-worker|image: repository-service-tuf-cli' ../../docker-compose.yml >> $GITHUB_OUTPUT
+        if [ "$GITHUB_OUTPUT" != "" ]; echo "Local image has been used in docker-compose.yml" exit 1; fi
+
```
This step should be executed after we checkout the code

MVrachev commented:
I am currently investigating why those CI workflow runs fail when locally they succeed.
Sorry for the many force pushes.

rdimitrov commented:
The reason is that the line is still there even though you've commented it 👍 

Let's try to change the search pattern in a way that works for us or if we cannot come up with a workaround maybe decide to not leave it commented 👍

MVrachev commented:
> The reason is that the line is still there even though you've commented it 👍

Maybe you have saw a previous version of the pr, but I have removed the comment a while ago.

The real problem it seems is with bash variables.
Because I wanted to make it more readable I wanted to move the result of the `egrep` command in a bash variable by trying maany things.
It seems that for some reason I had problems instantiating a variable because I tried what you (@rdimitrov) suggested but before I moved it to a one-liner it didn't work.

Thanks for the help @rdimitrov! I am happy to see it solved.

rdimitrov commented:
> > The reason is that the line is still there even though you've commented it 👍
> 
> Maybe you have saw a previous version of the pr, but I have removed the comment a while ago.
> 
> The real problem it seems is with bash variables. Because I wanted to make it more readable I wanted to move the result of the `egrep` command in a bash variable by trying maany things. It seems that for some reason I had problems instantiating a variable because I tried what you (@rdimitrov) suggested but before I moved it to a one-liner it didn't work.
> 
> Thanks for the help @rdimitrov! I am happy to see it solved.

Glad I helped! 👍 

I noticed that you've discarded the message you previously had in place, so feel free to update what we output in case the image is found 👍

kairoaraujo commented on .github/workflows/ci.yml:
```diff
@@ -16,6 +16,11 @@ jobs:
+
+    - name: Check if any local image is used in docker-compose.yml
+      run: |
+        if [[ "$(egrep -w 'image: repository-service-tuf-api|image: repository-service-tuf-worker|image: repository-service-tuf-cli' docker-compose.yml -c)" -ne "0" ]]; then echo "Found" && exit 1; fi
```
```suggestion
        egrep -w 'image:\s+repository-service-tuf-api|image:\s+repository-service-tuf-worker' docker-compose.yml
        if [[ $rc -eq "0" ]]; then \
          echo "Local image has been used in docker-compose.yml" \
          exit 1 \
        fi
```

- The `repository-service-cli` is not provided as a image and can be removed
- I added some extra in the regex to avoid more than one white space
- Instead of using count, we can use the shell return code (`$?`), which is the standard.
- neat: multiline for better reading

rdimitrov reviewed (approved):
Looks great 👍
